examples/threading/threading.c

```diff
@@ -13,8 +14,50 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+
+    if(thread_func_args == NULL)
+   {
+   	ERROR_LOG("Function Arguments are NULL\n");
+   	return NULL;	
+   }
+    // Waiting for certain time - custom sleep function
+    int wait_to_obtain_ms = thread_func_args->wait_to_obtain_ms;
+    char command1[100];
+    snprintf(command1, sizeof(command1), "./sleep.sh %d", wait_to_obtain_ms);
+    int delay1 = system(command1);
```

Comment:
@BhaktiRamani this is more complicated and probably less resilient than using the built in `sleep` or `usleep` function.

Also you should think about whether there's a chance this script should exit early and add handling for this case.